test/test_jit.py

```diff
@@ -2876,6 +2890,17 @@ def transpose(x):
             None, verbose=False)
         self.assertExpected(onnx_ish, subname='f2')
 
+    def test_shape_analysis_loop(self):
+        def foo(a, b, x):
```

Comment:
A comment for this test case? It's not clear which regression it is trying to prevent. Looks like it doesn't cover the "premature broadcasting" problem

Reply:
I added a note. It does test premature broadcasting since there is a broadcast in the first iteration of the loop but not in the second.